resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/internal/properties/Bug468PerformanceTest.java

```diff
@@ -158,9 +159,18 @@ public void test() throws CoreException {
 		}, this.project, IWorkspace.AVOID_UPDATE, new NullProgressMonitor());
 
 		long maxTime = TimeUnit.MILLISECONDS.convert(1, TimeUnit.MINUTES);
+		long maxTimeMacOS = TimeUnit.MILLISECONDS.convert(75, TimeUnit.SECONDS);
+
 		assertTrue("The expected min time(ms): " + 0 + ", actual time(ms): " + timeTakenForDeletingFiles[0],
 				0 <= timeTakenForDeletingFiles[0]);
-		assertTrue("The expected max time(ms): " + maxTime + ", actual time(ms): " + timeTakenForDeletingFiles[0],
-				timeTakenForDeletingFiles[0] <= maxTime);
+
+		if (OS.isWindows() || OS.isLinux()) {
```

Comment:
I would propose to only add a special branch for MacOS and put everything else into the else branch. In the unlikely case that there is some other OS than Windows, Linux and MacOS, nothing will be tested here.